Harvest.Web/Controllers/ProjectController.cs

```diff
@@ -31,19 +31,31 @@ public ActionResult Index()
             return View("React");
         }
 
-        // TODO: move or determine proper permissions
-        [Authorize(Policy = AccessCodes.FieldManagerAccess)]
         public async Task<ActionResult> Active()
         {
+            var user = await _userService.GetCurrentUser();
+            var userRoles = await _dbContext.Permissions
+                .Where(p => p.UserId == user.Id)
+                .Select(p => p.Role.Name)
+                .ToArrayAsync();
+            var hasNonPiRoles = userRoles.Any(r => r != Role.Codes.PI);
+            var isPi = userRoles.Contains(Role.Codes.PI);
+
             // TODO: only show projects where between start and end?
-            return Ok(await _dbContext.Projects.Include(p => p.PrincipalInvestigator).Where(p => p.IsActive).ToArrayAsync());
+            return Ok(await _dbContext.Projects
+                .Include(p => p.PrincipalInvestigator)
+                .Where(p => p.IsActive && (
+                    //PI's can only view their own projects
```

Comment:
I think a better way to do this would be to have a separate method for Active() and Mine().  One shows all projects you can see, one just shows where you are PI.  I think that's how Cal wanted it originally but we should think about it.  Also, please do some parenthesis when doing multiple conditionals like this.

Reply:
Splitting into Active() and Mine() means moving the check on PI to client-side. The fetch effects for both ProjectListContainer and ProjectSelection components would need to be updated.

Reply:
@srkirkland Do you mean have the ProjectListContainer.tsx call a different endpoint based on roles?

Reply:
Yes I agree

Reply:
@sprucely @jSylvestre let me clarify better -- no, the container wouldn't call different endpoints based on role, and no checks would be moved client side.  We would just have two different routes.  You would need to be supervisor or FM to visit the regular see all projects list.  Anyone else could see /projects/me and it would show projects where they are a PI.

Reply:
done